cloudinit/handlers/jinja_template.py

```diff
@@ -82,21 +94,26 @@ def render_jinja_payload_from_file(
     @return: A string of jinja-rendered content with the jinja header removed.
         Returns None on error.
     """
+    if detect_template(payload)[0] != "jinja":
+        raise NotJinjaError("Payload is not a jinja template")
```

Comment:
I think this exception should be handled in [cloudinit/cmd/devel/render.py::handle_args](https://github.com/canonical/cloud-init/pull/1722/files#diff-900ac554685fc30e9e71da127c37f66d08e6b7d10a64aa50c4dc42b71d6a68b3R56), otherwise the `render` command will raise it and exit with non 0.

Additionally, I think we should also unit-test cover the interaction between `util.read_conf`, `cmd.devel.render` and this exception.